Show share failure notification

igitur opened the pull request:
Fixes #371

igitur commented:
cc @gheja

gheja commented:
I think that looks great. Thank you!

cwackerfuss commented on src/lib/share.ts:
```diff
@@ -41,9 +42,19 @@ export const shareStatus = (
+      }
+    }
+  } catch (error) {
+    handleShareFailure()
```
Is this try/catch block still necessary with `handleShareFailure` also being caught on L51?

igitur replied:
@cwackerfuss I'll look into it soon.

igitur replied:
OK, so while testing on different browsers, I realised they don't all support exactly the same API. So rather than checking whether every every property and method is available, I just wrapped the entire block in a try/catch. For example (and this is just hypothetical), if a browser doesn't have `navigator.clipboard.writeText`, then an exception would occur, but I don't think the `.catch` on L51 will catch it. So I think we should keep this as a failsafe.

igitur commented:
This is still draft, because I want to add a new modal that is displayed when the Web Share API and clipboard methods fail. The user will then be able to manually copy the share text from a textarea element.

igitur commented:
On 2nd thought, I'll log a separate PR for that. Let's get this through and out in the wild first.

gheja commented:
Looks great, just one thing: the message disappears a bit too soon, I had to press the button 3 times to read the whole message. But of course, the player can do the same.

To emulate the failure I just did this in the console:
```javascript
navigator.clipboard.writeText = null
```

igitur commented:
> the message disappears a bit too soon

I added a new `LONG_ALERT_TIME_MS = 10000` variable in `settings.ts` and implemented this for this notification.
